rpcs3/Emu/RSX/Overlays/overlay_save_dialog.cpp

```diff
@@ -104,24 +105,59 @@ namespace rsx
 			m_description->back_color.a    = 0.f;
 			m_time_thingy->back_color.a    = 0.f;
 
+			fade_animation.duration = 0.15f;
+
 			return_code = selection_code::canceled;
 		}
 
+		void save_dialog::close(bool use_callback, bool stop_pad_interception)
+		{
+			fade_animation.current = color4f(1.f);
+			fade_animation.end = color4f(0.f);
+			fade_animation.active = true;
+
+			close_requested = get_system_time() + 150000;
+			close_use_callback = use_callback;
+			close_stop_pad_interception = stop_pad_interception;
+		}
+
 		void save_dialog::update()
 		{
 			m_time_thingy->set_text(date_time::current_time());
 			m_time_thingy->auto_resize();
+
+			const u64 t = get_system_time();
+			if (!creation_time)
+			{
+				// First tick
+				creation_time = t;
+				Emu.GetCallbacks().play_sound(fs::get_config_dir() + "sounds/snd_system_ok.wav");
+				return;
+			}
+
+			if (fade_animation.active)
+			{
+				fade_animation.update(rsx::get_current_renderer()->vblank_count);
+			}
+
+			if (close_requested && t >= close_requested)
+			{
```

Comment:
I'm not really sure if this is a good idea.
If the dialog is closed then it is closed.
Close can be called from different sources.
How do you know we're not killing it?

Reply:
I've added a check so that it can't be "closed twice."

Reply:
same issue.
Solely adding a counter does nothing to fix the issue.
You'll have to remove the override and simply issue the fade when the user closes the dialog

Reply:
aka in the input handler

Reply:
I meant the ``else if (!close_requested)`` in close(), which prevents the timer from being changed until the dialog is closed. Button presses are ignored while fade_animation.active == true.
But you're right, doing it in the input handler would be better, so I'll do that.